lib/liquid/static_registers.rb

```diff
@@ -1,36 +1,35 @@
 # frozen_string_literal: true
 
+require 'forwardable'
+
 module Liquid
   class StaticRegisters
+    extend Forwardable
+
     attr_reader :static, :registers
 
+    def_delegators :@cache, :[], :key?
+
     def initialize(registers = {})
       @static    = registers.is_a?(StaticRegisters) ? registers.static : registers
       @registers = {}
+
+      @cache = @static.dup
```

Comment:
Why bother having yet another cache.  If we are going to eagerly duplicate the static registers, then we could just do that for `@registers`

```suggestion
      @static = registers.is_a?(StaticRegisters) ? registers.static : registers
      @registers = @static.dup
```

However, if there are a lot of registers that aren't used in the rendered snippet, that could add extra overhead.  Was `Liquid::StaticRegisters` added to do this lazily for performance reasons?  Otherwise, why do we bother even having it in the first place when we could just store the static registers in the currently unused `Liquid::Context#static_registers` attribute

Reply:
The api expose an attr reader for :registers, I'm not sure of what would be the implications on the public contract...